pkg/lint/rules/chartfile.go

```diff
@@ -85,6 +87,18 @@ func validateChartNameDirMatch(chartDir string, cf *chart.Metadata) error {
 	return nil
 }
 
+func validateChartAPIVersion(cf *chart.Metadata) error {
+	if cf.APIVersion == "" {
```

Comment:
Could we somehow tie into [`cf.Validate()`](https://github.com/helm/helm/blob/55ae0c2efb48751cefe51cd66567608c0f7a650e/pkg/chart/metadata.go#L71) instead? That way we could keep the chart metadata validation DRY and in one package. 

Perhaps we could change the error output in `.Validate` to return typed errors like `ErrNoAPIVersionFound` or `ErrInvalidAPIVersion`. What do you think about that?

Reply:
I like the idea of typing into the `Validate` function in the chart package. From the looks of things, the current function is simple compared to what linting is doing on the `Chart.yaml` file. Other parts of linting have not been moved over to `Validate` either.

Should we do that migration in a separate PR? One that handles the change wholesale.

Reply:
Yeah, we can do that in a separate PR. 👍